web3/datastructures.py

```diff
@@ -114,10 +114,16 @@ def __delattr__(self, key: str) -> None:
         raise TypeError("This data is immutable -- create a copy instead of modifying")
 
     def __hash__(self) -> int:
-        return hash(tuple(sorted(tupleize_lists_nested(self).items())))
+        return hash(
+            tuple(
+                sorted(tupleize_lists_nested(self).items(), key=lambda item: str(item))
```

Comment:
I don't believe the addition of the `key` arg here is breaking. Previously, it would throw a `TypeError` if the  keys were not comparable. Every example of an `AttributeDict` I've found uses only strings as keys, but the class is explicitly typed with `TKey`, not `String`.